ReactWindows/ReactNative.Shared/UIManager/DependencyObjectExtensions.cs

```diff
@@ -14,8 +14,8 @@ namespace ReactNative.UIManager
     /// </summary>
     public static class DependencyObjectExtensions
     {
-        private static readonly ConditionalWeakTable<DependencyObject, DependencyObjectData> s_properties =
-            new ConditionalWeakTable<DependencyObject, DependencyObjectData>();
+        private static readonly ConcurrentDictionary<DependencyObject, DependencyObjectData> s_properties =
```

Comment:
>ConcurrentDictionary [](start = 32, length = 20)

Actually, after re-reviewing, I thought of the case where you tear down a React view from the root (e.g., by disposing the ReactInstanceManager or closing a secondary window). Wouldn't we have abandoned views in this case?

Reply:
I think you're right, I may just revert this.
When closing a window there seems to be some RemoveRootView that JS is sending with its last breath, but the enqueued item to NativeViewHierarchyManager is beaten by the dispatched OnDestroy, so..
And other scenarios are not really covered.

So I'll revert to the magic of "weak tables".

One more that comes to mind and I have no answer for (other than an ugly O(N) cleanup) is the _reactTagToOperationQueue map...

---
In reply to: [166569515](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166569515) [](ancestors = 166569515)